## Capítulo 1.1.: Modelos estadísticos para control de calidad

### Caso 1.1.1: Análisis de capacidad de proceso

In [2]:
import numpy as np
import math
import scipy.stats as st

#### Ejemplo 1:

Se tiene que en cierto proceso de fabricación de antibacteriales el porcentaje medio de alcohol en el gel es de 79,012% y su deviación estándar es de 0,545%. Se ha determinado que el valor nominal de la característica de calidad es de 80,00% y su discrepancia es 1,50%.

**a) Calcule la capacidad del proceso (Cp):**

R/ Se sabe que el índice de capacidad del proceso Cp esta dado por:
$$\rightarrow C_{p}=\frac{T}{6\sigma'}$$

Y:
$$\rightarrow T = 2\nabla = LSE - LIE$$

Donde:
* $\nabla$ es la discrepancia y de ahora en adelante se le asignará la letra 'D', 
* $\sigma'$ que es la desviación estándar se le asignará la letra 's',y 
* 'T' es la tolerancia. 

A continuación se procederá determinar la capacidad del proceso:

In [3]:
s = 0.00545
D = 0.015
T = 2*D

Cp = T/(6*s)
print(Cp)

0.9174311926605504


Dado que $0.67 < C_{p} < 1.00$ entonces esto por lo pronto indica que el proceso no tiene capacidad de fabricar producto dentro de las especificaciones, requiere modificaciones para generar producto que alcance una calidad satisfactoria.

**b) Determine los límites de especificación del proceso:**

R/ Se tiene que los límites de especificación del proceso están definidos como sigue:

$$\rightarrow LIE = VN - \nabla$$
$$\rightarrow LSE = VN + \nabla$$

Donde:

* VN es el valor nominal definido para el proceso, y
* $\nabla$ es la discrepancia.

Por tanto, los límites de especificación son:

In [4]:
VN = 0.8

LIE = np.round(VN - D,4)
LSE = np.round(VN + D,4)
print("LIE = "+ str(LIE) + " y LSE = " +str(LSE))

LIE = 0.785 y LSE = 0.815


**c) Calcule los límites de fluctuación natural del proceso:**

R/ Se tiene que los límites de fluctuación natural del proceso están dados por la siguiente fórmula:

$$\rightarrow LFN_{i,s} = \bar{X'} \pm 3\sigma'$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación estándar del proceso.

Entonces, los límites de fluctuación son:

In [5]:
X = 0.79012

LIFN = np.round(X - 3*s,4)
LSFN = np.round(X + 3*s,4)

print("LIFN = " + str(LIFN) + " y LSFN = " + str(LSFN))

LIFN = 0.7738 y LSFN = 0.8065


**d) ¿La fluctuación natural del proceso se encuentra dentro de los límites de especificación?**

R/ Se compara si LIFN es mayor o igual a LIE y si LSFN es menor o igual a LSE:

In [6]:
print(LIFN>=LIE, LSFN<=LSE)

False True


Se tiene que **el límite inferior fluctuación natural (LIFN) es mas bajo (<) que el límite inferior de especificación (LIE)**, por lo que se concluye que **el proceso está generando producto no conforme por debajo de dicho límite.**

**e)¿Cuál es la proporción de producto no conforme que se está generando por debajo del LEI?**

R/ Inicialmente se tiene que determinar el valor Z estandarizado para el límite de especificación inferior. Esto se obtiene con la siguiente fórmula:

$$\rightarrow Z_{inf}=\frac{LIE-\bar{X'}}{\sigma'}$$

Obteniendo el siguiente resultado:

In [7]:
Zinf = np.round((LIE - X)/s,4)
print(Zinf)

-0.9394


Luego de esto, se obtiene de la tabla de distribución normal estándar la probabilidad de obtener un valor Zinf menor o igual ($\leq$) a -0.9394, así:

In [8]:
P = np.round(st.norm.cdf(Zinf),4)
print(P)

0.1738


Es decir que la proporción de producto no conforme generado por debajo del límite de especificación es del 17.37%.

**f) ¿Entre que valores debería oscilar la media del proceso para que el porcentaje de no conformidad sea máximo 10%?**

Inicialmente se debe despejar $\bar{X'}$ de la fórmula de Z estandarizado inferior, así:

$$Z_{inf}=\frac{LIE-\bar{X'}}{\sigma'}\rightarrow\bar{X'}_{inf}=LIE-\sigma'Z_{inf}$$

Lo mismo ocurre con la formula de Z estandarizado superior:

$$Z_{sup}=\frac{LSE-\bar{X'}}{\sigma'}\rightarrow\bar{X'}_{sup}=LSE-\sigma'Z_{sup}$$

Donde:
* $Z_{inf}$ es el valor de Z estandarizado inferior que representa una probabilidad de 10% de obtener producto no conforme por debajo de LIE, y
* $Z_{sup}$ es el valor de Z estandarizado superior que representa una probabilidad de 90% (100% - 10%) de no obtener inconformidades de producto por encima de LSE.

Por lo tanto:

In [9]:
VO_inf = 0.10
Zinf_vo = np.round(st.norm.ppf(VO_inf),4)


VO_sup = 1 - VO_inf
Zsup_vo = np.round(st.norm.ppf(VO_sup),4)

print("Zinf = "+str(Zinf_vo) + " y Zsup = " + str(Zsup_vo))

Zinf = -1.2816 y Zsup = 1.2816


Entonces, los valores entre los cuales puede oscilar $\bar{X'}$ para obtener a lo sumo 10% de producto no conforme son:

In [10]:
Xinf_vo = np.round(LIE - (Zinf_vo*s),4) # Cuando el proceso entrega 10% de producto no conforme por debajo del límite inferior de especificación.
Xsup_vo = np.round(LSE - (Zsup_vo*s),4) # Cuando el proceso entrega 10% de producto no conforme por encima del límite superior de especificación.
print("Xinf = " + str(Xinf_vo) + " y Xsup = " + str(Xsup_vo))

Xinf = 0.792 y Xsup = 0.808
